internal/webserver/server_test.go

```diff
@@ -0,0 +1,109 @@
+//
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package webserver
+
+import (
+	"testing"
+
+	"net/http"
+	"net/http/httptest"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	logger "github.com/edgexfoundry/go-mod-core-contracts/clients/logging"
+	"github.com/gorilla/mux"
+	"github.com/stretchr/testify/assert"
+)
+
+var lc logger.LoggingClient
+
+func init() {
+	lc = logger.NewClient("app_functions_sdk_go", false, "./test.log", "DEBUG")
+}
+
+func TestConfigureAndPingRoute(t *testing.T) {
+
+	webserver := WebServer{
+		LoggingClient: lc,
+	}
+	webserver.ConfigureStandardRoutes()
+
+	req, _ := http.NewRequest("GET", clients.ApiPingRoute, nil)
+	rr := httptest.NewRecorder()
+	webserver.router.ServeHTTP(rr, req)
+
+	body := rr.Body.String()
+	assert.Equal(t, "pong", body)
+
+}
+
+func TestConfigureAndConfigRoute(t *testing.T) {
+
+	webserver := WebServer{
+		LoggingClient: lc,
+	}
+	webserver.ConfigureStandardRoutes()
+
+	req, _ := http.NewRequest("GET", clients.ApiConfigRoute, nil)
+	rr := httptest.NewRecorder()
+	webserver.router.ServeHTTP(rr, req)
+
+	expected := `{"Writable":{"SubscribeTopic":"","PublishTopic":"","LogLevel":""},"Logging":{"EnableRemote":false,"File":""},"Registry":{"Host":"","Port":0,"Type":""},"Service":{"BootTimeout":0,"CheckInterval":"","ClientMonitor":0,"Host":"","Port":0,"Protocol":"","StartupMsg":"","ReadMaxLimit":0,"Timeout":0},"MessageBus":{"Host":"","Port":0,"Type":""},"Binding":{"Type":"","Name":"","Topic":""}}` + "\n"
+	body := rr.Body.String()
+	assert.Equal(t, expected, body)
+}
+
+func TestConfigureAndMetricsRoute(t *testing.T) {
+
+	webserver := WebServer{
+		LoggingClient: lc,
+	}
+	webserver.ConfigureStandardRoutes()
+
+	req, _ := http.NewRequest("GET", clients.ApiMetricsRoute, nil)
+	rr := httptest.NewRecorder()
+	webserver.router.ServeHTTP(rr, req)
+
+	body := rr.Body.String()
+	assert.NotNil(t, body, "Metrics not populated")
+}
```

Comment:
How about also testing that the data can be marshaled and isn't all zeros?

Reply:
The metrics JSON string is now unmarshaled into an object and tested for a non-zero value